### PR TITLE
split cbi scenario and reduce fixture scope

### PR DESCRIPTION
**Pull request by ocefpaf**
This change should fix the failure (or at least make reveal a bug :smile:) that @lsetiawan observed in https://github.com/ODM2/WOFpy/pull/177#issuecomment-326120952

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,11 +12,15 @@ matrix:
   - python: 2.7
     env: TEST_TARGET=cli
   - python: 2.7
+    env: TEST_TARGET=build_cbi
```
Comment (ocefpaf) [inline .travis.yml]: We should run tests that require an external connection in the `allow_failures` section so we can debug them quickly.

Note to self. The current setup is not smart, it runs a lot of test in both scenarios. I'll figure out what depends on `build_cbi` in another PR and optimize this.

File: .travis.yml [hunk 2]
```diff
@@ -48,9 +48,9 @@ script:
       py.test -s -rxs -v ;
     fi
 
-  - if [[ $TEST_TARGET == 'default' ]]; then
+  - if [[ $TEST_TARGET == 'build_cbi' ]]; then
       python wof/examples/flask/cbi/build_cbi_cache.py ;
-      python wof/examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db ;
+      python wof/examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db > /dev/null & py.test -s -rxs -v ;
```
Comment (lsetiawan) [inline .travis.yml]: @ocefpaf I don't think this is a part of CBI.

File: test/test_cli.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from wof.wofpy_config import makedirs
 
 
-@pytest.fixture
+@pytest.fixture(scope='function')
```
Comment (ocefpaf) [inline test/test_cli.py]: I could not reproduce that locally nor on `Travis-CI`, but reducing the scope of the fixture is recommended anyways, so here it goes.

File: .travis.yml (hunk 2 above)
Comment (ocefpaf) [reply .travis.yml]: I though it used stuff from the step above. I can optimize this in my next commit. Hold on...

File: .travis.yml [hunk 4]
```diff
@@ -39,15 +43,18 @@ install:
   - python setup.py sdist && version=$(python setup.py --version) && pushd dist  && pip install WOFpy-${version}.tar.gz && popd
 
 before_script:
-# Commented out due to external service not working. 8/30/17. lsetiawan.
-#  - python wof/examples/flask/cbi/build_cbi_cache.py
   - python wof/examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db &
```
Comment (ocefpaf) [inline .travis.yml]: @lsetiawan this made me realize that all the `cbi` tests are skipped anyways :smile:

See https://travis-ci.org/ocefpaf/WOFpy/jobs/270205139#L915

Is that something we should look into?

Comment (lsetiawan) [reply .travis.yml]: @ocefpaf Nah. Not for this release. And it's not really regarding ODM2. Someone else can pick it up 😜

Comment (ocefpaf) [reply .travis.yml]: Cool. So I'll make another commit to put all `cbi` related tests in the new scenario. That way we get less noise on the `default` scenario. Hold on...

**Comment by ocefpaf** [comment]
@lsetiawan this is ready for a review now.

**Pull request merged**